### PR TITLE
Recommend setting IsTrimmable in library projects

### PR DESCRIPTION
**Pull request by eerhardt**
## Summary

Recommend setting IsTrimmable in library projects

### CODE REVIEW

File: docs/core/deploying/prepare-libraries-for-trimming.md [hunk 1]
```diff
@@ -26,11 +26,13 @@ These instructions show how to enable and resolve static analysis warnings to pr
...
 
-The Roslyn analyzer is useful for a fast feedback cycle with IDE integration, but is currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It is important to follow the next steps to ensure that your library is fully compatible with trimming.
+Setting `<IsTrimmable>true</IsTrimmable>` will also enable a Roslyn analyzer. The Roslyn analyzer is useful for a fast feedback cycle with IDE integration, but is currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It is important to follow the next steps to ensure that your library is fully compatible with trimming.
```
Comment (sbomer) [inline docs/core/deploying/prepare-libraries-for-trimming.md:33]:
```suggestion
Setting `<IsTrimmable>true</IsTrimmable>` will also enable a Roslyn analyzer for trim compatibility. The Roslyn analyzer is useful for a fast feedback cycle with IDE integration, but is currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It is important to follow the next steps to ensure that your library is fully compatible with trimming.
```

**Review by sbomer — APPROVED** [review]
Thank you!

**Review by vitek-karas — APPROVED** [review]

File: docs/core/deploying/prepare-libraries-for-trimming.md [hunk 2]
```diff
@@ -26,11 +26,13 @@ These instructions show how to enable and resolve static analysis warnings to pr
...
 
-The Roslyn analyzer is useful for a fast feedback cycle with IDE integration, but is currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It is important to follow the next steps to ensure that your library is fully compatible with trimming.
+Setting `<IsTrimmable>true</IsTrimmable>` will also enable a Roslyn analyzer for trim compatibility. The Roslyn analyzer is useful for a fast feedback cycle with IDE integration, but is currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It is important to follow the next steps to ensure that your library is fully compatible with trimming.
```
Comment (adegeo) [inline docs/core/deploying/prepare-libraries-for-trimming.md:33]: - Question: When the following paragraph says "it's important to follow the **next steps**..." whare are the **next steps**?

- Suggestion: Using a more active voice:

```suggestion
Setting `<IsTrimmable>true</IsTrimmable>` enables a Roslyn analyzer for trim compatibility. The Roslyn analyzer is useful for quick feedback in your IDE, but it's currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It's important to follow the next steps to ensure that your library is fully compatible with trimming.
```

File: docs/core/deploying/prepare-libraries-for-trimming.md [hunk 3]
```diff
@@ -26,11 +26,13 @@ These instructions show how to enable and resolve static analysis warnings to pr
 
 Ensure you are using the .NET 6 SDK for these steps. They will not work correctly in previous versions.
 
-## Enable Roslyn analyzer
+### Set IsTrimmable
 
-Set `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` (in .NET 6+) in your library project. This will not have any effect on the output, but it will enable trim analysis during build via a Roslyn analyzer.
+Set `<IsTrimmable>true</IsTrimmable>` (in .NET 6+) in your library project. This will mark your assembly as "trimmable". Being trimmable means when your library is used in a trimmed application the assembly can have its unused members trimmed in the final output.
```
Comment (adegeo) [inline docs/core/deploying/prepare-libraries-for-trimming.md:31]: What do people use for .NET 5 or .NET Core 3.1 (if available)?

**Review by adegeo — CHANGES_REQUESTED** [review]

File: docs/core/deploying/prepare-libraries-for-trimming.md (hunk 3 above)
Comment (eerhardt) [reply docs/core/deploying/prepare-libraries-for-trimming.md:31]: This functionality isn't available in 5 or earlier. It is only available in 6.

File: docs/core/deploying/prepare-libraries-for-trimming.md (hunk 2 above)
Comment (eerhardt) [reply docs/core/deploying/prepare-libraries-for-trimming.md]: The next steps are in the next section:

`### Show all warnings`

File: docs/core/deploying/prepare-libraries-for-trimming.md [hunk 4]
```diff
@@ -26,11 +26,13 @@ These instructions show how to enable and resolve static analysis warnings to pr
...
 
-The Roslyn analyzer is useful for a fast feedback cycle with IDE integration, but is currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It is important to follow the next steps to ensure that your library is fully compatible with trimming.
+Setting `<IsTrimmable>true</IsTrimmable>` enables a Roslyn analyzer for trim compatibility. The Roslyn analyzer is useful for quick feedback in your IDE, but it's currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It's important to follow the next steps to ensure that your library is fully compatible with trimming.
```
Comment (adegeo) [inline docs/core/deploying/prepare-libraries-for-trimming.md:33]:
```suggestion
Setting `<IsTrimmable>true</IsTrimmable>` enables a Roslyn analyzer for trim compatibility. The Roslyn analyzer is useful for quick feedback in your IDE, but it's currently incomplete. It doesn't cover all trim analysis warnings, but the set of patterns it understands will improve over time to give more complete coverage. The Roslyn analyzer also isn't able to analyze the implementations of reference assemblies that you depend on. It's important to follow the steps outlined in the rest of this article to ensure that your library is fully compatible with trimming.
```

**Review by adegeo — APPROVED** [review]

**Pull request merged**